app/controllers/api/v2/courses_controller.rb

```diff
@@ -55,7 +65,20 @@ def index
       end
 
       def show
-        render jsonapi: @course, include: params[:include]
+        # https://github.com/jsonapi-rb/jsonapi-rails/issues/113
+        render jsonapi: @course, include: params[:include], class: {
+          Course: SerializableCourse,
+          SiteStatus: SerializableSiteStatus,
+          Site: SerializableSite,
+          Subject: SerializableSubject,
+          PrimarySubject: SerializableSubject,
+          SecondarySubject: SerializableSubject,
+          ModernLanguagesSubject: SerializableSubject,
+          FurtherEducationSubject: SerializableSubject,
+          Provider: SerializableProvider,
+          ProviderEnrichment: SerializableProviderEnrichment,
+          RecruitmentCycle: SerializableRecruitmentCycle,
```

Comment:
This is becoming a real bore. I wonder if we shouldn't have something that automatically registers serializers, for example:

```ruby
module API::V2
  SERIALIZERS = {}

  class SerializableCourse
    # Taking the liberty to assume we can assign to a constant Hash. Is true?
    API::V2::SERIALIZERS[Course] = self
    ...
  end
end
```

And then here:

```ruby
      def show
        render jsonapi: @course, include: params[:include], class: API::V2::SERIALIZERS
      end
```

Reply:
Automatically registering serializers might start us on the rabbithole of introducing magic which makes it difficult for us to change things later. Additionally, a given model may have multiple serializers (e.g: ProviderSuggestion vs Provider). Where the hash of classes is repeated we can simply use a service to provide them.